release_docs/RELEASE.txt

```diff
@@ -38,6 +38,8 @@ New features and changes
 ========================
     Configuration:
     -------------
+    - Removed support for autotools build system.
```

Comment:
Might also want to have another line where we mention that bin/cmakehdf4 was removed. That probably should have been done in a separate PR since it's not really an Autotools thing.